### DISCUSSION TITLE
DSi Camera support! Feedback wanted!

### DISCUSSION THREAD

**Discussion by rafaelvcaetano** [rafaelvcaetano/melonDS-android#1093] [General]
Hello guys!

Work on the support for the DSi camera has started, and it's working at its bare minimum. There are still a lot of things let that need to be implemented for the feature to work properly but, the most important thing, which is being able to use the device's camera, seems to be OK for the most part!

![image](https://user-images.githubusercontent.com/9086379/216144581-1f81ae65-9e7e-40fb-97d6-4f7da5467931.png)

# But I want your help!

Given the large amount of camera configurations out there, what I have implemented is bound to crash on a lot of devices. So, I would like you to test this feature on your devices and let me know if something isn't working properly! If that's the case, post here the following information:

* Device model
* Screen orientation
* Camera being used (either front or back camera. If you are not sure, try testing using the DSi's system camera feature. It uses the front camera by default)

You can download a build with the DSi camera support from the [nightly builds](https://github.com/rafaelvcaetano/melonDS-android/actions/workflows/main.yaml).

**Comment by PiixelNerd** [comment]
My device: Galaxy S20, snapdragon 865, android 12
Notes: DSi bios doesn't start anymore, in previous versions it did! 

Another question out of this topic, was there any improvement in not starting DSiware from the files anymore? 
I thought it was so good not having to put them on the Nand. xD

**Reply by rafaelvcaetano** [reply]
The DSiWare Manager should be working fine on the latest nightly release. A couple of issues were found and addressed. Are you still having problems using it?

**Reply by PiixelNerd** [reply]
I'm using the latest version (the one with the camera function)...
In this version the DSi bios won't start, making it impossible to test the camera function!

About DSiWare games, in the past I could run them from the list of games, now I can't anymore due to the change you made, which prevents starting them from the list.
I just wanted to know if this brought improvements in any way, because I particularly liked being able to run DSiWare through the list of roms without having to install it on the Nand.

**Comment by KostaSaizo7** [comment]
I haven't tested this  yet but I want to point out an issue. In official melonDS for Windows, there is an issue where enabling the camera in certain games slows the emulator to single digit numbers. See https://github.com/melonDS-emu/melonDS/issues/1548 > Low FPS possibly due to DSP when enabling DSi Camera

**Comment by GokiPotato** [comment]
redmi note 8t, dsi firmware crashes immediately in this camera version of melonds

**Comment by Gonckzilla** [comment]
Hey there, before my report, I wanted to say thank you for your work, I love this emulator, being able to boot the system menu and having a dsiware manager is awesome. I look up to people like you and hope to be able to support your work more in the future, if I get to be a successful adult.

Device is Realme GT 5G Snapdragon 888, Android 12.
I intended on testing this build with the dsiware Photo Dojo, but found the app crashing right after trying to boot the DSi firmware. Do let me know if there's any logs or any other testing I can provide, I'll gladly do so!

**Comment by christopher3242** [comment]
I think redmi pro 10

bug green look

**Comment by hdk2d** [comment]
Already tested it works well
Look https://youtu.be/zpQiuPwoPco
Information in video description

Edit update:
I tried all the filters and it works excellent and I have no problems.

**Comment by hdk2d** [comment]
Test other Device Snap 8 Gen 1
Motorola Edge 30 Pro Android 13™ with MyUX
![Screenshot_20230907-124659.png](https://github.com/rafaelvcaetano/melonDS-android/assets/86883038/a0c71cc6-f2d6-4db9-9849-9db31de6b0b8)

**Comment by MilesTailsFan2005** [comment]
It just crashes for me
Basically I get a green screen then crash with the option - Physical Cameras 
Phone - HONOR X7

**Comment by DolphinOfficial** [comment]
Instant crashing whenever the camera is used.
Phone: Pixel 8 Pro
I would give video evidence, but blame GitHub.

**Comment by Cicoria1917** [comment]
OPPO A98
Front camera, back camera crashes immediately with green screen
Vertical screen orientation.

**Comment by hdk2d** [comment]
work on Redmi K70 with HyperOS
specs:
CPU Snap 8 gen 2
Ram:16gb
